src/huggingface_hub/inference/_providers/__init__.py

```diff
@@ -0,0 +1,67 @@
+from typing import Any, Dict, Optional, Protocol, Union
+
+from .fal_ai import text_to_image as falai_text_to_image
+from .hf_inference import conversational as hf_inference_conversational
+from .hf_inference import text_to_image as hf_inference_text_to_image
+from .replicate import text_to_image as replicate_text_to_image
+from .sambanova import conversational as sambanova_conversational
+from .together import conversational as together_conversational
+from .together import text_generation as together_text_generation
+from .together import text_to_image as together_text_to_image
+
+
+class TaskProviderHelper(Protocol):
+    """Protocol defining the interface for task-specific provider helpers."""
+
+    def build_url(self, model: Optional[str] = None) -> str: ...
+    def map_model(self, model: Optional[str] = None) -> str: ...
+    def prepare_headers(self, headers: Dict, **kwargs) -> Dict: ...
```

Comment:
```suggestion
    def prepare_headers(self, headers: Dict, *, token: Optional[str] = None) -> Dict: ...
```

for now the only kwarg I've seen is `token` in `fal_ai`. Therefore I think we can reduce the scope of what `prepare_headers` can accept. If we need more kwargs in the future, we can always add them later.